packages/yew/src/dom_bundle/dom_position.rs

```diff
@@ -0,0 +1,215 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use web_sys::Node;
+
+#[derive(Clone)]
+pub struct RetargetableDomPosition {
+    target: Rc<RefCell<DomPosition>>,
+}
+
+#[derive(Clone)]
+enum DomPositionVariant {
+    Node(Option<Node>),
+    Chained(RetargetableDomPosition),
+}
```

Comment:
```suggestion
#[derive(Clone)]
enum DomPositionVariant<'a> {
    Node(Option<&'a Node>),
    Chained(RetargetableDomPosition),
}
```

I think DomPosition is never stored so it doesn't need to live `'static`.
It would be nice to have this as a reference as cloning a JsValue actually creates JavaScript calls to create a new spot on the ref array so it's relatively expensive to clone them.

Reply:
The issue with this is that `DomPosition` is returned from the various `ReconcileTarget::*` trait methods, without a lifetime readily availabe. I'll see what can be done.